src/systems/waypoint-system.js

```diff
@@ -323,7 +329,24 @@ export class WaypointSystem {
       this.waitOneTick = false;
       return;
     }
+
+    const hashUpdated = window.location.hash !== "" && this.previousWaypointHash !== window.location.hash;
+
+    if (hashUpdated && this.initialSpawnHappened) {
```

Comment:
I introduced `initialSpawnHappened` as once you enter the scene, async, `moveToSpawnPoint` will get called and it will override the hash location that will be set otherwise on the first tick after detecting the new hash